event/datadeal/data_delivery_event.go

```diff
@@ -0,0 +1,236 @@
+package datadeal
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	datadealtypes "github.com/medibloc/panacea-core/v2/x/datadeal/types"
+	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
+	"github.com/medibloc/panacea-doracle/config"
+	"github.com/medibloc/panacea-doracle/crypto"
+	"github.com/medibloc/panacea-doracle/event"
+	"github.com/medibloc/panacea-doracle/panacea"
+	log "github.com/sirupsen/logrus"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+type DataDeliveryVoteEvent struct {
+	reactor event.Reactor
+}
+
+func NewDataDeliveryVoteEvent(s event.Reactor) DataDeliveryVoteEvent {
+	return DataDeliveryVoteEvent{s}
+}
+
+func (e DataDeliveryVoteEvent) GetEventType() string {
+	return "data_delivery"
+}
+
+func (e DataDeliveryVoteEvent) GetEventAttributeKey() string {
+	return "vote_status"
+}
+
+func (e DataDeliveryVoteEvent) GetEventAttributeValue() string {
+	return "'started'"
+}
+
+func (e DataDeliveryVoteEvent) EventHandler(event ctypes.ResultEvent) error {
+
+	dealIDStr := event.Events[datadealtypes.EventTypeDataDeliveryVote+"."+datadealtypes.AttributeKeyDealID][0]
+	dataHash := event.Events[datadealtypes.EventTypeDataDeliveryVote+"."+datadealtypes.AttributeKeyDataHash][0]
+
+	dealID, err := strconv.ParseUint(dealIDStr, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	oraclePrivKey := e.reactor.OraclePrivKey()
+
+	voteOption, deliveredCid, err := e.verifyAndGetVoteOption(dealID, dataHash, oraclePrivKey)
+	if err != nil {
+		log.Errorf("error while verify: %v", err)
+		if voteOption == oracletypes.VOTE_OPTION_UNSPECIFIED {
+			return err
+		}
+	}
+
+	msgVoteDataDelivery, err := e.makeDataDeliveryVote(
+		e.reactor.OracleAcc().GetAddress(),
+		dataHash,
+		deliveredCid,
+		dealID,
+		voteOption,
+		oraclePrivKey.Serialize(),
+	)
+	if err != nil {
+		return err
+	}
+
+	txBuilder := panacea.NewTxBuilder(*e.reactor.QueryClient())
+
+	txBytes, err := e.generateTxBytes(msgVoteDataDelivery, e.reactor.OracleAcc().GetPrivKey(), e.reactor.Config(), txBuilder)
+	if err != nil {
+		return err
+	}
+
+	if err := e.broadcastTx(e.reactor.GRPCClient(), txBytes); err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+func (e DataDeliveryVoteEvent) verifyAndGetVoteOption(dealID uint64, dataHash string, oraclePrivKey *btcec.PrivateKey) (oracletypes.VoteOption, string, error) {
+
+	dataSale, err := e.reactor.QueryClient().GetDataSale(dealID, dataHash)
+	if err != nil {
+		if errors.Is(err, datadealtypes.ErrDataSaleNotFound) {
+			return oracletypes.VOTE_OPTION_NO, "", err
```

Comment:
This case, we print `error` type log.
But I think if oracle does vote to no, we shouldn''t printing `error`  type log.

What I meant was to separate the logs into `info` types and `error` types.

Reply:
Ah, I see. So I fixed it in [f95f622](https://github.com/medibloc/panacea-doracle/pull/93/commits/f95f622a0e79c9323ce203410674f9fce0f2e42c). Is is okay?

Reply:
Actually, when error occurs in `EventHandler`, doracle prints error log.
https://github.com/medibloc/panacea-doracle/blob/db3efc002fba6fae94742ae0fe455578c8f2318b/event/subscriber.go#L59

Thus, I suggest to log with the details of data sale, such as deal ID and data hash.